llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -421,27 +466,132 @@ Expected<SectionRef> SectionRef::create(MCCASBuilder &MB,
   return get(B->build());
 }
 
+Expected<SmallVector<char, 0>> MCCASReader::applyRelocationAddends(
+    const SmallVector<cas::ObjectRef> &Refs, StringRef RelocData,
+    StringLiteral KindString, StringRef SectionName,
+    bool &IsPaddingRefPresent) {
+  if (auto E = decodeRelocations(*this, RelocData))
+    return std::move(E);
+  SmallVector<char, 0> SectionData;
+  auto RelocRef = MCObjectProxy::get(Schema, Schema.CAS.getProxy(Refs[0]));
+  if (!RelocRef)
+    return RelocRef.takeError();
+  if (auto R = RelocAddendRef::Cast(*RelocRef)) {
+    auto RelocationData = R->getData();
+    for (uint32_t I = 1; I < Refs.size(); I++) {
```

Comment:
We can greatly simplify the logic in this loop if we compute something like `IsLastRefPadding` _outside_  the loop. With this, most error handling can be folded into a single error. And then we do this:

`for (auto Ref : Refs.drop_front().drop_back(IsLastRefPadding))`